DatadogSDKBridge/Classes/Implementation/DdLogsImplementation.swift

```diff
@@ -16,17 +16,18 @@ internal protocol NativeLogger {
 }
 
 public class DdLogsImplementation: DdLogs {
-    private let logger: NativeLogger
+    private lazy var logger: NativeLogger = loggerProvider()
+    private let loggerProvider: () -> NativeLogger
```

Comment:
How about adding extra test for lazy initialization? We could assert that the `provider` block is only called once:

```swift
func testItInitializesNativeLoggerOnlyOnce() {
    let expectation = self.expectation(description: "Initialize logger once")

    let logger = DdLogsImplementation { [unowned self] in
        expectation.fulfill() // this will fail if fulfilled more than once
        return self.mockNativeLogger
    }

    // When
    (0..<10).forEach { _ in logger.debug(message: "foo", context: [:]) }

    // Then
    waitForExpectations(timeout: 0.5, handler: nil)
}
```